main.go

```diff
@@ -99,7 +103,29 @@ func (c *Controller) handleNewCSR(key string) error {
 		return nil
 	}
 
-	// TODO and CSR checking logic here
+	var machineListObj runtime.Object
+	err = c.clientset.RESTClient().Get().Namespace("openshift-cluster-api").Name("machines").Do().Into(machineListObj)
```

Comment:
should we approve if the machine API isn't available?

Reply:
we can check the username `system:node[s]...`  other than that, i don't think there is much we can check.